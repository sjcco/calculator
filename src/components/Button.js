import PropTypes from 'prop-types';

const Button = props => {
  const {
    name,
    btnColor,
    clickHandler,
    wide,
  } = props;
  return (
    <>
      <button type="button" className={wide ? `button ${btnColor} zero` : `button ${btnColor}`} onClick={() => clickHandler(name)}>
        { name }
      </button>
    </>
  );
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  btnColor: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
  wide: PropTypes.bool.isRequired,
};

export default Button;
