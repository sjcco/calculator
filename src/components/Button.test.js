import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import Button from './Button';

test('Element renders correctly', () => {
  const mockHandler = jest.fn();
  const component = renderer.create(<Button wide={false} btnColor="bg-gray" name="7" clickHandler={mockHandler} />).toJSON();
  expect(component).toMatchSnapshot();
});

test('buttons fire events', () => {
  const mockHandler = jest.fn();
  render(<Button wide={false} btnColor="bg-gray" name="7" clickHandler={mockHandler} />);
  fireEvent.click(screen.getByText(/7/));
  expect(mockHandler).toHaveBeenCalledTimes(1);
});
