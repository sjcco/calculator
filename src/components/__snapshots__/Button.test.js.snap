// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Element renders correctly 1`] = `
<button
  className="button bg-gray"
  onClick={[Function]}
  type="button"
>
  7
</button>
`;
